libraries/node-core-library/src/Text.ts

```diff
@@ -68,6 +75,13 @@ export class Text {
     return input.replace(Text._newLineRegEx, '\n');
   }
 
+  /**
+   * Converts all newlines in the provided string to use the default newline type for this operating system.
+   */
+  public static convertToOsDefault(input: string): string {
```

Comment:
The API name `Text.convertToOsDefault(x)` does not clearly indicate what is being converted.  (Tabs? Text encoding?) 
 Whereas with `Text.convertToLf(x)` it was somewhat clear that newlines are being converted, if people recognize that `CrLf` and `Lf` are newlines.

One idea would be to call this `Text.convertToOsNewline(x)`. Then the 3 variations would be:

- `Text.convertToCrLf(x)`
- `Text.convertToLf(x)`
- `Text.convertToOsNewline(x)`


However I suspect that `NewlineKind.OsDefault` will be rarely used (see other note about deterministic output).  So perhaps the third API should be a parameterized version:

- `Text.convertToCrLf(x)`
- `Text.convertToLf(x)`
- `Text.convertTo(x, NewlineKind.OsDefault)`

This API would also eliminate the need for the private helper [FileSystem._convertLineEndings](https://github.com/microsoft/rushstack/blob/b9466acc57ca58c5ccb63bfe299cbc32cc8e182e/libraries/node-core-library/src/FileSystem.ts#L523).

Reply:
And the switch statement in [JsonFile.updateString](https://github.com/microsoft/rushstack/blob/b9466acc57ca58c5ccb63bfe299cbc32cc8e182e/libraries/node-core-library/src/JsonFile.ts#L131).